torch/onnx/utils.py

```diff
@@ -1280,9 +1278,15 @@ def _track_module_attributes_forward_hook(module, input, output):
             m.register_forward_hook(_track_module_attributes_forward_hook)
             m.register_forward_pre_hook(_track_module_attributes_forward_pre_hook)
 
+    trace_module_map = {
+        _m: torch._C._jit_onnx_create_full_scope_name(torch.typename(type(_m)), _n)
+        for _n, _m in model.named_modules()
+    }
+    torch.jit._trace._trace_module_map = trace_module_map
     if isinstance(export_modules_as_functions, bool) and export_modules_as_functions:
-        trace_module_map = __setup_trace_module_map()
-        export_modules_as_functions = {v for k, v in trace_module_map.items()}
+        export_modules_as_functions = {
+            torch.typename(type(_m)) for _m, _ in trace_module_map.items()
```

Comment:
```suggestion
            torch.typename(type(named_module)) for named_module in trace_module_map
```

maybe this for simplicity?